processing/src/main/java/io/druid/segment/LongDimensionIndexer.java

```diff
@@ -40,7 +40,7 @@ public Long processRowValsToUnsortedEncodedKeyComponent(Object dimValues)
       throw new UnsupportedOperationException("Numeric columns do not support multivalue rows.");
     }
 
-    return DimensionHandlerUtils.convertObjectToLong(dimValues);
+    return DimensionHandlerUtils.convertObjectToLong(dimValues, true);
```

Comment:
If we get an unparseable string value here, and reportParseExceptions is false, I think it'd be better to replace it with zero than to skip the entire row. That way we aren't throwing away as much data.

If reportParseExceptions is true then yes, we should throw an exception out.

(+ similar comment for other numeric types)